crates/next-api/src/project.rs

```diff
@@ -1918,17 +1972,40 @@
         should_trace,
     );
 
-    let full = ModuleGraph::from_graphs(vec![base_single_module_graph, additional_module_graph]);
+    let full_with_unused_references =
+        ModuleGraph::from_graphs(vec![base_single_module_graph, additional_module_graph])
+            .to_resolved()
+            .await?;
+
+    let full = if turbopack_remove_unused_imports {
+        full_with_unused_references
+            .without_unused_references(
+                *compute_binding_usage_info(full_with_unused_references, true)
```

Comment:
We could make this more efficient by initializing the variables with the binding usage info of the base graph. So we quickly skip existing entries and only really traverse the new entry.